source/slang/slang-diagnostic-defs.h

```diff
@@ -274,6 +274,7 @@ DIAGNOSTIC(31006, Error, attributeFunctionNotFound, "Could not find function '$0
 DIAGNOSTIC(31100, Error, unknownStageName, "unknown stage name '$0'")
 DIAGNOSTIC(31101, Error, unknownImageFormatName, "unknown image format '$0'")
 DIAGNOSTIC(31101, Error, unknownDiagnosticName, "unknown diagnostic '$0'")
+DIAGNOSTIC(31102, Error, nonPositiveNumThreads, "numThreads attribute or local_size_x,y,z layout must be positive integers '$0'")
```

Comment:
Note that the HLSL attribute name is `numthreads` (all lower case). I'd also argue we shouldn't bother listing the `local_size_...` cases here. I might suggest:

```
expected a positive integer in 'numthreads' attribute, got '$0'
```